### PR TITLE
chore: Add timeline to remove deprecated resourceCustomizations

### PR DESCRIPTION
**Pull request by svghadi**
**What type of PR is this?**
/kind chore

**What does this PR do / why we need it**:
This PR adds removal timeline for deprecated `resourceCustomizations`. The `resourceCustomizations` was deprecated in `v0.5.0` and is planned to be removed in `v0.8.0`.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:
1. Run the operator locally using make install run
2. Create a argocd instance using deprecated `resourceCustomizations` field
   > eg: https://argocd-operator.readthedocs.io/en/latest/reference/argocd/#resource-customizations-example
3. Execute `kubectl get events | grep ResourceCustomizations`
4. Observe that deprecation event was emitted
   ```
   9m19s       Warning   DeprecationNotice         argocd/example-argocd                               ResourceCustomizations is deprecated, and support will be removed in Argo CD Operator v0.8.0/OpenShift GitOps v1.10.0. Please use the new formats `ResourceHealthChecks`, `ResourceIgnoreDifferences`, and `ResourceActions`.
   ```

### CODE REVIEW

File: controllers/argocd/configmap.go [hunk 1]
```diff
@@ -414,7 +414,7 @@ func (r *ReconcileArgoCD) reconcileArgoConfigMap(cr *argoprojv1a1.ArgoCD) error
 
 		// Emit event providing users with deprecation notice for ResourceCustomization if not emitted already
 		if currentInstanceEventEmissionStatus, ok := DeprecationEventEmissionTracker[cr.Namespace]; !ok || !currentInstanceEventEmissionStatus.ResourceCustomizationsDeprecationWarningEmitted {
-			err := argoutil.CreateEvent(r.Client, "Warning", "Deprecated", "ResourceCustomizations is deprecated, please use the new formats `ResourceHealthChecks`, `ResourceIgnoreDifferences`, and `ResourceActions` instead.", "DeprecationNotice", cr.ObjectMeta, cr.TypeMeta)
+			err := argoutil.CreateEvent(r.Client, "Warning", "Deprecated", "ResourceCustomizations is deprecated, and support will be removed in Argo CD Operator v0.7.0/OpenShift GitOps v1.10.0. Please use the new formats `ResourceHealthChecks`, `ResourceIgnoreDifferences`, and `ResourceActions`.", "DeprecationNotice", cr.ObjectMeta, cr.TypeMeta)
```
Comment (jaideepr97) [inline controllers/argocd/configmap.go:417]: @svghadi 0.7.0 is the next release of the operator
GitOps v1.10 corresponds to v0.8.0 of the operator

File: docs/reference/argocd.md [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ spec:
...
 
 !!! warning 
-    `resourceCustomizations` is being deprecated so is encouraged to use `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions` instead. It is the user's responsibility to not provide conflicting resources if they choose to use both methods of resource customizations. 
+    `resourceCustomizations` is being deprecated, and support will be removed in Argo CD Operator v0.7.0 so is encouraged to use `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions` instead. It is the user's responsibility to not provide conflicting resources if they choose to use both methods of resource customizations.
```
Comment (jaideepr97) [inline docs/reference/argocd.md:1011]: `0.7.0` => `0.8.0`

File: docs/reference/argocd.md [hunk 3]
```diff
@@ -1149,7 +1149,7 @@ resource.customizations.actions.apps_Deployment: |
 ### Resource Customizations Example
 
 !!! warning 
-    `resourceCustomizations` is being deprecated in favor of `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions`.
+    `resourceCustomizations` is being deprecated, and support will be removed in Argo CD Operator v0.7.0. Please use the new formats `resourceHealthChecks`, `resourceIgnoreDifferences`, and `resourceActions`.
```
Comment (jaideepr97) [inline docs/reference/argocd.md:1152]: `0.7.0` =>` 0.8.0`

File: controllers/argocd/configmap.go (hunk 1 above)
Comment (svghadi) [reply controllers/argocd/configmap.go]: Updated

File: docs/reference/argocd.md (hunk 2 above)
Comment (svghadi) [reply docs/reference/argocd.md]: Updated

File: docs/reference/argocd.md (hunk 3 above)
Comment (svghadi) [reply docs/reference/argocd.md]: Updated

**Review by wtam2018 — APPROVED** [review]

**Review by jaideepr97 — APPROVED** [review]
/lgtm
thanks @svghadi

**Pull request merged**